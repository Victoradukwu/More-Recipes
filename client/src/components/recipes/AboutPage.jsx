import React from 'react';

const AboutPage = () => (
    <div>
    <h4>This a test About page</h4>
    <p> You are welcome to my About page bla bla bla ...</p>
    </div>
  );
export default AboutPage;