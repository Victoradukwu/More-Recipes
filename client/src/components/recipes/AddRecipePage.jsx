import React, { Component } from 'react';
import { connect } from 'react-redux';

class AddRecipe extends Component {
  constructor(props){
    super(props);
  }
  render() {
    console.log(this.props) 
    return (
      <div className="card mx-auto text-center w-50">
        <div className="card-header" id = "cardHeader">
          <h4>Add a recipe</h4>
        </div>
    
        <div className="card-body">
          <div className = "forminfo" style = {{color: 'blue'}}>
            To submit a recipe to the catalog, fill out the form below and submit.
          </div>
          <br/><br/>
          <form action="">                    
            <fieldset className="form-group">
              <label htmlFor="name">Name of recipe:</label>
              <input type="text" className="form-control form-input" id="name" name="name"/>
            </fieldset>
            <fieldset className="form-group">
              <label htmlFor="name">Category:</label>
              <input type="text" className="form-control form-input" id="Category" name="Category"/>
            </fieldset>
            <fieldset className="form-group">
              <label htmlFor="ingredients">Ingredients:</label><br/>>
              <textarea nameName="ingredients" id="ingredients" cols="30" rows="7" className="form-control form-input" />
            </fieldset>                    
            <fieldset className="form-group">
              <label htmlFor="steps">Instructions:</label><br/>>
              <textarea nameName="instructions" id="instructions" cols="30" rows="7" className="form-control form-input" />
            </fieldset>                    =
            <button type="submit" className="btn" style = {{ backgroundColor: '#336600', color: 'white' }}>Submit Recipe</button>
          </form>
        </div>
      </div>
        );
  };
}


export default AddRecipe;