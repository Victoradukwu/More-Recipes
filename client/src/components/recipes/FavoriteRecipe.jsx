import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const FavoriteRecipe = ({ userFavorite }) => (
  <tr>
    <td> {userFavorite.Recipe.recipeName}</td>
    <td>{userFavorite.category}</td>
    <td>
      <Link
        to={`/recipeDetail/${userFavorite.Recipe.id}`}
        className="btn search"
      >Check Details
      </Link>
    </td>

  </tr>
);

FavoriteRecipe.propTypes = {
  userFavorite: PropTypes.object.isRequired
};

export default FavoriteRecipe;
