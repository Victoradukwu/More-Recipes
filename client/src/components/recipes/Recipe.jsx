import { Link } from 'react-router-dom';
import React from 'react';
import PropTypes from 'prop-types';

const Recipe = ({ recipe }) => (
  <div className="card-holder">
    <div className="card bg-light text-black">
      <img
        className="card-img"
        src={recipe.recipePicture || 'http://res.cloudinary.com/victoradukwu/image/upload/v1520512629/hd1.jpg'}
        alt="Sample recipe"
        style={{ height: '250px' }}
      />
      <p
        className="card-text mt-0"
        style={{ backgroundColor: 'white', padding: '10px' }}
      >
        <span
          className="fa fa-thumbs-up fa-lg "
          style={{ float: 'left' }}
        >{recipe.upvote}
        </span>
        <span
          className="fa fa-thumbs-down fa-lg"
          style={{ float: 'right' }}
        >{recipe.downvote}
        </span>
      </p>

      <div className="card-body">
        <h4 className="card-title">{recipe.recipeName}</h4>

        <Link
          to={`/recipeDetail/${recipe.id}`}
          className="btn search"
        >Check Details
        </Link>
      </div>
    </div>
  </div>
);

Recipe.propTypes = {
  recipe: PropTypes.shape({
    id: PropTypes.number,
    recipeName: PropTypes.string,
    category: PropTypes.string,
    ingredients: PropTypes.string,
    instructions: PropTypes.string,
    upvote: PropTypes.number,
    downvote: PropTypes.number,
    views: PropTypes.number,
    favorites: PropTypes.number,
    createdAt: PropTypes.any,
    updatedAt: PropTypes.any,
    userId: PropTypes.number,
    reviews: PropTypes.array,
    User: PropTypes.shape({ id: PropTypes.number, name: PropTypes.string })
  }).isRequired,
};
Recipe.defaultProps = {

};

export default Recipe;
