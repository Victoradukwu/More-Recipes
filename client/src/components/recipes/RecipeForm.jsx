import React from 'react';
import PropTypes from 'prop-types';
import TextInput from '../common/TextInput';
// import ImageInput from '../common/ImageInput';

const RecipeForm = ({ recipe, onChange, errors, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <TextInput
      name="recipeName"
      label="Recipe Name"
      value={recipe.recipeName}
      onChange={onChange}
      error={errors.recipeName}
    />
    <TextInput
      name="category"
      label="Category"
      value={recipe.category}
      onChange={onChange}
      error={errors.caregory}
    />
    <TextInput
      name="ingredients"
      label="Ingredients"
      value={recipe.ingredients}
      onChange={onChange}
      error={errors.ingredients}
    />
    <TextInput
      name="instruction"
      label="Instruction"
      value={recipe.instruction}
      onChange={onChange}
      error={errors.instruction}
    />
    {/* <ImageInput
      name="recipePicture"
      label="RecipePicture"
      value={recipe.recipePicture}
      onChange={onChange}
      error={errors.recipePicture}
    /> */}
    <button
      type="submit"
      className="btn"
      style={{ backgroundColor: '#336600', color: 'white' }}
    >
      Submit Recipe
    </button>


  </form>
);

RecipeForm.propTypes = {
  recipe: PropTypes.object,
  onChange: PropTypes.func.isRequired,
  errors: PropTypes.objectOf(PropTypes.string),
  onSubmit: PropTypes.func.isRequired
};

RecipeForm.defaultProps = {
  recipe: {},
  errors: {}
};

export default RecipeForm;
