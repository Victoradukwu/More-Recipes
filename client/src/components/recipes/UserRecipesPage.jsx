import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchUserRecipes, deleteRecipe } from '../../actions/recipeActions';
import RecipeList from './RecipeList';

class UserRecipesPage extends Component {
  constructor(props) {
    super(props);
    this.redirectToCreateRecipePage = this.redirectToCreateRecipePage.bind(this);
    this.isSignedIn = this.isSignedIn.bind(this);
    this.deleteRecipe = this.deleteRecipe.bind(this);
  }
  componentDidMount() {
    this.props.fetchUserRecipes();
  }
  redirectToCreateRecipePage() {
    this.props.history.push('/recipe');
  }

  isSignedIn() {
    if (localStorage.token === undefined) {
      this.props.history.push('/signin');
    }
  }
  deleteRecipe(event) {
    event.preventDefault();
    this.props.actions.deleteRecipe(id);
  }
  render() {
    this.isSignedIn();
    return (
      <div>
        <br /><br />
        <div className="row">
          <div className="col-sm-2" />
          <div className="col-sm-8">
            <h3>Your Recipes</h3>
            <p style={{ display: 'inline-block', float: 'left' }}>Click on a recipe name to edit</p>
            <button
              className="btn btn-success btn-lg"
              style={{ display: 'inline-block', float: 'right' }}
              onClick={this.redirectToCreateRecipePage}
            >
                Create a recipe
            </button>
            {
              this.props.userRecipes &&
              <RecipeList
                userRecipes={this.props.userRecipes}
                deleteRecipe ={this.props.deleteRecipe}
              />
            }
          </div>
          <div className="col-sm-2" />
        </div>
        <br /><br />

      </div>
    );
  }
}

UserRecipesPage.propTypes = {
  fetchUserRecipes: PropTypes.func.isRequired,
  userRecipes: PropTypes.array.isRequired,
  history: PropTypes.any.isRequired,
  deleteRecipe: PropTypes.func
};

const mapStateToProps = state => ({
  userRecipes: state.userRecipes
});

export default connect(mapStateToProps, { fetchUserRecipes, deleteRecipe })(UserRecipesPage);
