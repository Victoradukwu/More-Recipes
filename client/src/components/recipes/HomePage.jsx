import React, { Component } from 'react';
import { connect } from 'react-redux';
import ReactPaginate from 'react-paginate';
import PropTypes from 'prop-types';

import Recipe from './Recipe';

import { fetchRecipes } from '../../actions/recipeActions';

class HomePage extends Component {
  constructor(props) {
    super(props);
    this.handlePageClick = this.handlePageClick.bind(this);

    this.state = { recipes: [] };
  }

  componentDidMount() {
    this.props.fetchRecipes('/api/v1/recipes?sort=upvotes&order=des');
  }

  handlePageClick(data) {
    const selected = data.selected + 1;
    this.props.fetchRecipes(`/api/v1/recipes?sort=upvotes&order=des&page=${selected}`);
  }

  render() {
    return (
      <div>
        <div className="container-fluid">
          <img alt="" src={require('../../assets/img/img001.jpeg')} style={{ width: '100%', height: '620px' }} />
        </div>
        <h4>
          <small>Our top picks</small>
        </h4>
        <hr />
        <div className="container">
          <div className="container bg-3 text-center main-content">
            <div className="grid-holder">
              {this.props.payload.recipes.map(recipe => (
                <Recipe key={recipe.id} recipe={recipe} />
          ))}
            </div>
          </div>
          <br />

          <ReactPaginate
            previousLabel="&laquo;"
            nextLabel="&raquo;"
            breakClassName="break-me"
            pageCount={this.props.payload.pages}
            marginPagesDisplayed={2}
            pageRangeDisplayed={5}
            onPageChange={this.handlePageClick}
            containerClassName="pagination pagination-lg justify-content-center"
            subContainerClassName="page-item"
            activeClassName="active"
          />
          <br /><br />
          <br /><br />
        </div>
      </div>
    );
  }
}
HomePage.propTypes = {
  fetchRecipes: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  isloading: state.fetchRecipeRequest,
  hasErrored: state.fetchRecipeFailure,
  payload: state.fetchRecipeSuccess
});

const mapDispatchToProps = dispatch => ({
  fetchRecipes: url => dispatch(fetchRecipes(url))
});
export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
