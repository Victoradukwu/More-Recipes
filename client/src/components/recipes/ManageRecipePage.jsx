import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import axios from 'axios';
import * as recipeActions from '../../actions/recipeActions';
import RecipeForm from './RecipeForm';
import validate from '../../helpers/validate';
import checkImageFile from '../../helpers/checkImageFile';
import setAuthorizationToken from '../../helpers/setAuthorizationToken';

require('dotenv').config();

/**
 * @description
 * @class ManageRecipePage
 * @extends {Component}
 */
class ManageRecipePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      recipe: Object.assign({}, props.recipe),
      errors: {},
      defaultImgSrc: '../../assets/img/hd8.jpg',
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.handleImageChange = this.handleImageChange.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.error.status) {
      this.setState({ errors: nextProps.error });
    }
    if (this.props.recipe.id !== nextProps.recipe.id) {
      this.setState({ recipe: Object.assign({}, nextProps.recipe) });
    }
  }
  onChange(event) {
    const field = event.target.name;
    const myRecipe = Object.assign({}, this.state.recipe);
    myRecipe[field] = event.target.value;
    return this.setState({ recipe: myRecipe });
  }

  async onSubmit(event) {
    event.preventDefault();
    const recipeObject = {
      recipeName: event.target.recipeName.value,
      category: event.target.category.value,
      ingredients: event.target.ingredients.value,
      instructions: event.target.instructions.value,
      id: this.state.recipe.id
    };
    if (this.state.recipe.recipePicture) {
      const recipeImage = await this.getImgURL();
      const imageUrl = recipeImage.data.url;
      recipeObject.recipePicture = imageUrl;
    }
    this.setState({ errors: {} });

    setAuthorizationToken(localStorage.getItem('token'));
    const { isValid, errors } = validate(recipeObject);
    if (isValid) {
      this.setState({ errors: {} });
      this.props.actions.submitRecipe(recipeObject)
        .then(() => {
          this.props.history.push('/myRecipes');
        });
    }
    this.setState({ errors });
  }

  getImgURL() {
    const imgPix = this.state.recipe.recipePicture;
    delete axios.defaults.headers.common['x-access-token'];
    const imageData = new FormData();
    imageData.append('file', imgPix);
    imageData.append('upload_preset', process.env.CLOUDINARY_UPLOAD_PRESET);
    return axios.post(process.env.CLOUDINARY_URL, imageData);
  }

  handleImageChange(event) {
    event.persist();
    if (event.target.files && event.target.files[0]) {
      const file = event.target.files[0];
      const filereader = new FileReader();
      checkImageFile(filereader, file, (fileType) => {
        if (fileType === 'image/png' || fileType === 'image/gif' ||
          fileType === 'image/jpeg') {
          this.setState({
            recipe:
            Object.assign({}, this.state.recipe, { recipePicture: file })
          });

          filereader.onload = (e) => {
            this.setState({ defaultImgSrc: e.target.result });
          };
          filereader.readAsDataURL(file);
        } else {
          toastr.error('image must be in png, jpeg or gif format');
        }
      });
    } else {
      this.setState({ defaultImgSrc: '../../assets/img/hd8.jpg' });
    }
  }

  isSignedIn() {
    if (localStorage.token === undefined) {
      this.props.history.push('/signin');
    }
  }
  render() {
    this.isSignedIn();
    return (
      <div className="row">
        <div className="col-sm-3" />
        <div className="col-sm-6">
          <h3> Add/Edit Recipe</h3>
          <RecipeForm
            recipe={this.state.recipe}
            errors={this.state.errors}
            onSubmit={this.onSubmit}
            onChange={this.onChange}
            onImageUpload={this.handleImageChange}
            defaultImgSrc={this.state.defaultImgSrc}
          />
        </div>
        <div className="col-sm-3" />
      </div>
    );
  }
}
ManageRecipePage.propTypes = {
  error: PropTypes.object,
  history: PropTypes.shape({
    push: PropTypes.func
  }).isRequired,
  actions: PropTypes.object.isRequired,
  recipe: PropTypes.shape({
    id: PropTypes.number,
    recipeName: PropTypes.string,
    category: PropTypes.string,
    ingredients: PropTypes.string,
    instructions: PropTypes.string,
    upvote: PropTypes.number,
    downvote: PropTypes.number,
    views: PropTypes.number,
    favorites: PropTypes.number,
    createdAt: PropTypes.any,
    updatedAt: PropTypes.any,
    userId: PropTypes.number,
    reviews: PropTypes.array,
    User: PropTypes.shape({ id: PropTypes.number, name: PropTypes.string })
  }).isRequired,
};

ManageRecipePage.defaultProps = {
  error: {}
};
const getRecipeById = (recipes, id) => {
  const singleRecipe = recipes.filter(recipe =>
    parseInt(recipe.id, 10) === parseInt(id, 10));
  if (singleRecipe.length > 0) {
    return singleRecipe[0];
  }
  return 'Recipe Not Found';
};
const mapStateToProps = (state, ownProps) => {
  const recipeId = ownProps.match.params.id;
  let recipe = { recipeName: '', instructions: '', category: '' };
  if (recipeId && state.userRecipes.length > 0) {
    recipe = getRecipeById(state.userRecipes, recipeId);
  }
  return { recipe };
};
const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(recipeActions, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ManageRecipePage);
