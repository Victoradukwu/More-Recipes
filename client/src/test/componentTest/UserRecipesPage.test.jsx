import React from 'react';
import { shallow } from 'enzyme';
import { UserRecipesPage } from '../../components/recipes/UserRecipesPage';

const props = {
  fetchUserRecipes: jest.fn(),
  userRecipes: [{ recipeName: 'theOneRecipe' }],
  history: {
    push: jest.fn()
  },
  deleteRecipe: jest.fn(),
};

describe('UserRecipesPage', () => {
  it('matches snapshot and render page element properly', () => {
    const wrapper = shallow(<UserRecipesPage {...props} />);
    expect(wrapper.find('RecipeList').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
