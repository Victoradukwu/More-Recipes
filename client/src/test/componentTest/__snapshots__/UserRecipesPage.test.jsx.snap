// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRecipesPage matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserRecipesPage
    deleteRecipe={[MockFunction]}
    fetchUserRecipes={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    history={
      Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/signin",
            ],
          ],
        },
      }
    }
    userRecipes={
      Array [
        Object {
          "recipeName": "theOneRecipe",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <br />,
        <br />,
        <div
          className="row"
        >
          <div
            className="col-sm-2"
          />
          <div
            className="col-sm-8"
          >
            <h3>
              Your Recipes
            </h3>
            <p
              style={
                Object {
                  "display": "inline-block",
                  "float": "left",
                }
              }
            >
              Click on a recipe name to edit
            </p>
            <button
              className="btn search btn-lg"
              onClick={[Function]}
              style={
                Object {
                  "display": "inline-block",
                  "float": "right",
                }
              }
            >
              Create a recipe
            </button>
            <RecipeList
              deleteRecipe={[Function]}
              userRecipes={
                Array [
                  Object {
                    "recipeName": "theOneRecipe",
                  },
                ]
              }
            />
          </div>
          <div
            className="col-sm-2"
          />
        </div>,
        <br />,
        <br />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-sm-2"
            />,
            <div
              className="col-sm-8"
            >
              <h3>
                Your Recipes
              </h3>
              <p
                style={
                  Object {
                    "display": "inline-block",
                    "float": "left",
                  }
                }
              >
                Click on a recipe name to edit
              </p>
              <button
                className="btn search btn-lg"
                onClick={[Function]}
                style={
                  Object {
                    "display": "inline-block",
                    "float": "right",
                  }
                }
              >
                Create a recipe
              </button>
              <RecipeList
                deleteRecipe={[Function]}
                userRecipes={
                  Array [
                    Object {
                      "recipeName": "theOneRecipe",
                    },
                  ]
                }
              />
            </div>,
            <div
              className="col-sm-2"
            />,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "col-sm-2",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Your Recipes
                </h3>,
                <p
                  style={
                    Object {
                      "display": "inline-block",
                      "float": "left",
                    }
                  }
                >
                  Click on a recipe name to edit
                </p>,
                <button
                  className="btn search btn-lg"
                  onClick={[Function]}
                  style={
                    Object {
                      "display": "inline-block",
                      "float": "right",
                    }
                  }
                >
                  Create a recipe
                </button>,
                <RecipeList
                  deleteRecipe={[Function]}
                  userRecipes={
                    Array [
                      Object {
                        "recipeName": "theOneRecipe",
                      },
                    ]
                  }
                />,
              ],
              "className": "col-sm-8",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Your Recipes",
                },
                "ref": null,
                "rendered": "Your Recipes",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Click on a recipe name to edit",
                  "style": Object {
                    "display": "inline-block",
                    "float": "left",
                  },
                },
                "ref": null,
                "rendered": "Click on a recipe name to edit",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Create a recipe",
                  "className": "btn search btn-lg",
                  "onClick": [Function],
                  "style": Object {
                    "display": "inline-block",
                    "float": "right",
                  },
                },
                "ref": null,
                "rendered": "Create a recipe",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "deleteRecipe": [Function],
                  "userRecipes": Array [
                    Object {
                      "recipeName": "theOneRecipe",
                    },
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "col-sm-2",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <br />,
          <br />,
          <div
            className="row"
          >
            <div
              className="col-sm-2"
            />
            <div
              className="col-sm-8"
            >
              <h3>
                Your Recipes
              </h3>
              <p
                style={
                  Object {
                    "display": "inline-block",
                    "float": "left",
                  }
                }
              >
                Click on a recipe name to edit
              </p>
              <button
                className="btn search btn-lg"
                onClick={[Function]}
                style={
                  Object {
                    "display": "inline-block",
                    "float": "right",
                  }
                }
              >
                Create a recipe
              </button>
              <RecipeList
                deleteRecipe={[Function]}
                userRecipes={
                  Array [
                    Object {
                      "recipeName": "theOneRecipe",
                    },
                  ]
                }
              />
            </div>
            <div
              className="col-sm-2"
            />
          </div>,
          <br />,
          <br />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-sm-2"
              />,
              <div
                className="col-sm-8"
              >
                <h3>
                  Your Recipes
                </h3>
                <p
                  style={
                    Object {
                      "display": "inline-block",
                      "float": "left",
                    }
                  }
                >
                  Click on a recipe name to edit
                </p>
                <button
                  className="btn search btn-lg"
                  onClick={[Function]}
                  style={
                    Object {
                      "display": "inline-block",
                      "float": "right",
                    }
                  }
                >
                  Create a recipe
                </button>
                <RecipeList
                  deleteRecipe={[Function]}
                  userRecipes={
                    Array [
                      Object {
                        "recipeName": "theOneRecipe",
                      },
                    ]
                  }
                />
              </div>,
              <div
                className="col-sm-2"
              />,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "col-sm-2",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Your Recipes
                  </h3>,
                  <p
                    style={
                      Object {
                        "display": "inline-block",
                        "float": "left",
                      }
                    }
                  >
                    Click on a recipe name to edit
                  </p>,
                  <button
                    className="btn search btn-lg"
                    onClick={[Function]}
                    style={
                      Object {
                        "display": "inline-block",
                        "float": "right",
                      }
                    }
                  >
                    Create a recipe
                  </button>,
                  <RecipeList
                    deleteRecipe={[Function]}
                    userRecipes={
                      Array [
                        Object {
                          "recipeName": "theOneRecipe",
                        },
                      ]
                    }
                  />,
                ],
                "className": "col-sm-8",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Your Recipes",
                  },
                  "ref": null,
                  "rendered": "Your Recipes",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Click on a recipe name to edit",
                    "style": Object {
                      "display": "inline-block",
                      "float": "left",
                    },
                  },
                  "ref": null,
                  "rendered": "Click on a recipe name to edit",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Create a recipe",
                    "className": "btn search btn-lg",
                    "onClick": [Function],
                    "style": Object {
                      "display": "inline-block",
                      "float": "right",
                    },
                  },
                  "ref": null,
                  "rendered": "Create a recipe",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "deleteRecipe": [Function],
                    "userRecipes": Array [
                      Object {
                        "recipeName": "theOneRecipe",
                      },
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "col-sm-2",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
